Assignment4/AndriiLata - 230762/task07.py

```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+"""Task07.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1b-Nw-guQPOtdHzXpgaXShLYR8PfyRz-o
+
+**Task 07: Querying RDF(s)**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2023-2024/master/Assignment4/course_materials"
+
+"""First let's read the RDF file"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example6.rdf", format="xml")
+
+"""**TASK 7.1: List all subclasses of "LivingThing" with RDFLib and SPARQL**"""
+
+q1 = f"""SELECT ?subclass
+        WHERE {{ ?subclass rdfs:subClassOf ns:LivingThing. }} """
+for r in g.query(q1):
+  print(r.subclass)
+
+"""**TASK 7.2: List all individuals of "Person" with RDFLib and SPARQL (remember the subClasses)**
+
+"""
+
+q2 = f"""SELECT ?individuals
+        WHERE {{ ?individuals a ns:Person . }}"""
+
```

Comment:
not taking into account subclasses of :Person! (and instances of their subclases, and so on)